data-preprocessing/webtextbook_parsing_script/extract_textbook_from_url_notebook_.ipynb

```diff
@@ -0,0 +1,462 @@
+    "\n",
+    "\n",
+    "def remove_items(a, b):\n",
+    "    # this function removes a subset from a supperset of a list.\n",
+    "    # it was used to filter out Paaragraph_Not_in_Section\n",
+    "    # a = supperset = referring to all paragraphs (Para) , b = subset = referring to all sectioned paragraphs\n",
+    "    a = [item for item in a if item not in b]\n",
+    "    return a\n",
+    "\n",
+    "#---------------------------------------------------------------#\n",
+    "\n",
+    "#-----------Function: Extract table of contents -----------------#\n",
+    "import requests\n",
+    "from bs4 import BeautifulSoup\n",
+    "\n",
+    "import os\n",
+    "from urllib.parse import urljoin\n",
+    "import json\n",
+    "\n",
+    "def extract_rawTable_of_content(link, homePage):\n",
+    "\n",
+    "    # --> homePage = First landing pages of online view of the textbook(eg: \"1-introduction\")\n",
+    "    # ---> link = url of the site excluding landing page indexing (eg: \"https://openstax.org/books/university-physics-volume-3/pages/\")\n",
+    "\n",
+    "    website_link = link+homePage\n",
+    "    url_list = []\n",
+    "    \n",
+    "    # Send a GET request to the website\n",
+    "    response = requests.get(website_link)\n",
+    "    if response.status_code == 200:\n",
+    "        # Parse the HTML content using BeautifulSoup\n",
+    "        soup = BeautifulSoup(response.content, 'html.parser')\n",
+    "        \n",
+    "        # Find the table of contents div using its class\n",
+    "        table_of_contents_div = soup.find('div')\n",
+    "        \n",
+    "        if table_of_contents_div:\n",
+    "            # Find all the <a> tags within the table of contents div\n",
+    "            a_tags = table_of_contents_div.find_all('a')\n",
+    "            \n",
+    "            # Extract the href attribute from each <a> tag and store it in the list\n",
+    "            for a_tag in a_tags:\n",
+    "                href = a_tag.get('href')\n",
+    "                url_list.append(href)\n",
+    "\n",
+    "\n",
+    "                # if href.startswith(\"h\"):\n",
+    "            \n",
+    "        else:\n",
+    "            print(\"Table of contents div not found on the website.\")\n",
+    "                \n",
+    "    return unique(url_list) \n",
+    "#----------------------------------------------------------------#\n",
+    "\n",
+    "#----------- Extract URL for each page of the textbook -----------#\n",
+    "\n",
+    "def extract_url(link, pageList, maxNmber, char = []):\n",
+    "    # This function generates the dirst number or alphabet that\n",
+    "    # is part of the table of content requied\n",
+    "    #------------------------------------------------------------------------#\n",
+    "\n",
+    "    #-> maxNumber = maximum number of the index numbers of the table of content\n",
+    "    #--> char =  alphabets or string index in the table of content\n",
```

Comment:
will suggest using a different name as `char` already means something specific in programming languages so may be confusing to use here